packages/sites/src/get-members.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  IHubRequestOptions,
+  getPortalUrl,
+  Logger,
+  batch
+} from "@esri/hub-common";
+
+// TODO: once the Hub API User Search is complete, integrate
+// it in this function for AGO users, and fallback to the
+// current implementation for enterprise users
+
+/**
+ * Fetches and returns members given a list of usernames
+ *
+ * NOTE: AGO's user search endpoint is only available to
+ * authenticated users; however, since unauthenticated users
+ * should still be able to access public profiles, we support
+ * fetching members for both unauthenticated/authenticated
+ * users in this function
+ *
+ * @param {string[]} usernames List of usernames to search for
+ * @param {object} requestOptions IHubRequestOptions
+ */
+export function getMembers(
+  usernames: string[],
+  requestOptions: IHubRequestOptions
+): Promise<IUser[]> {
+  return requestOptions.authentication
+    ? authenticatedGetMembers(usernames, requestOptions)
+    : unauthenticatedGetMembers(usernames, requestOptions);
+}
+
+/**
+ * groups the provided usernames in chunks of 100 and creates query
+ * strings to batch fetch those members from AGO's user search
+ * (/community/users) endpoint. AGO will only return the subset of
+ * members which the current user has access to.
+ *
+ * @param {Array} usernames List of usernames to search for
+ * @param {object} requestOptions IHubRequestOptions
+ */
+function authenticatedGetMembers(
+  usernames: string[],
+  requestOptions: IHubRequestOptions
+): Promise<IUser[]> {
+  const urlPath = `${getPortalUrl(
+    requestOptions
+  )}/sharing/rest/community/users`;
+  const chunkSize = 100;
+  const chunkedUsernames = [];
+
+  for (let i = 0; i < usernames.length; i += chunkSize) {
+    chunkedUsernames.push(usernames.slice(i, i + chunkSize));
+  }
+
+  const chunkedOptions = chunkedUsernames.map(chunk => {
+    const q = chunk.map(username => `username:${username}`).join(" OR ");
+    return {
+      urlPath,
+      requestOptions: { params: { q, num: chunk.length }, ...requestOptions }
+    };
+  });
+
+  return batch(chunkedOptions, batchMemberRequest).then(batchedMembers => {
+    return batchedMembers.reduce((flat: IUser[], toFlatten: IUser[]) => {
+      return flat.concat(toFlatten);
+    }, []);
+  });
+}
+
+/**
+ * fetch members individually from AGO's /community/users/{username}
+ * endpoint. This endpoint, unlike the users search endpoint which
+ * only returns the subset of members that the current user has acces
+ * to, will only limit the information returned for each member
+ * (i.e. firstname, lastname and fullname will be empty strings if
+ * an unauthenticated user tries to access a non-public profile).
+ *
+ * @param {Array} usernames List of usernames to search for
+ * @param {object} requestOptions IHubRequestOptions
+ */
+function unauthenticatedGetMembers(
+  usernames: string[],
+  requestOptions: IHubRequestOptions
+): Promise<IUser[]> {
+  return Promise.all(
```

Comment:
I suppose there's some upper limit to the number of requests we can throw at `Promise.all()`, but I guess we can cross that bridge when we come to it.